framework/helpers/BaseHtml.php

```diff
@@ -1384,8 +1385,8 @@ private static function normalizeMaxLength($model, $attribute, &$options)
             unset($options['maxlength']);
             $attrName = static::getAttributeName($attribute);
             foreach ($model->getActiveValidators($attrName) as $validator) {
-                if ($validator instanceof StringValidator && $validator->max !== null) {
-                    $options['maxlength'] = $validator->max;
+                if ($validator instanceof StringValidator && ($validator->max !== null || $validator->length !== null)) {
+                    $options['maxlength'] = max($validator->max, $validator->length);
```

Comment:
Won't work because of array to string conversion and ability to specify min value only in `length`. See https://github.com/yiisoft/yii2/blob/master/framework/validators/StringValidator.php#L26

Reply:
You mean this convertation

https://github.com/yiisoft/yii2/blob/6a04ac9651fe33d264ca913cfa7fffd884f9e86b/framework/validators/StringValidator.php#L81-L89

is not work?

Reply:
You're right. The code is OK. Sorry.